### PR TITLE
Server attachment waiters

### PR DESCRIPTION
**Pull request by vicentepinto98**

### CODE REVIEW

File: services/iaasalpha/wait/wait.go [hunk 1]
```diff
@@ -165,3 +176,59 @@ func DeleteServerWaitHandler(ctx context.Context, a APIClientInterface, projectI
 	handler.SetTimeout(20 * time.Minute)
 	return handler
 }
+
+// ProjectRequestWaitHandler will wait for a request to succeed
+// It receives a request id that can be obtained from the x-request-id header in the http response of any operation in the IaaS API.
+// To get the raw http response of an operation using the SDK, use the runtime.WithCaptureHTTPResponse method from the core pkg.
+// Then the value of the request id can be obtained by accessing the header key which is defined in the constant XRequestIDHeader in this package.
+// Example usage:
```
Comment (vicentepinto98) [inline services/iaasalpha/wait/wait.go:184]: Not sure if this is too much in a method description. We will also have examples but I thought about making it clear here how to use everything together

Comment (joaopalet) [reply services/iaasalpha/wait/wait.go:185]: I agree it makes sense to have this detailed description :+1:

File: services/iaasalpha/wait/wait.go [hunk 2]
```diff
@@ -165,3 +177,114 @@ func DeleteServerWaitHandler(ctx context.Context, a APIClientInterface, projectI
...
+//
+//	var httpResp *http.Response
+//	ctxWithHTTPResp := runtime.WithCaptureHTTPResponse(context.Background(), &httpResp)
+//	err = iaasalphaClient.AddPublicIpToServer(ctxWithHTTPResp, projectId, serverId, publicIpId).Execute()
+//	requestId := httpResp.Header[wait.XRequestIDHeader][0]
+//
+// _, err = wait.ProjectRequestWaitHandler(context.Background(), iaasalphaClient, projectId, requestId).WaitWithContext(context.Background())
+func ProjectRequestWaitHandler(ctx context.Context, a APIClientInterface, projectId, requestId string) *wait.AsyncActionHandler[iaasalpha.Request] {
```
[the remark this answers is not in the document]
Comment (vicentepinto98) [reply services/iaasalpha/wait/wait.go]: Thanks, nice suggestion

File: services/iaasalpha/wait/wait.go [hunk 3]
```diff
@@ -165,3 +177,119 @@ func DeleteServerWaitHandler(ctx context.Context, a APIClientInterface, projectI
...
+//	var httpResp *http.Response
+//
+//	ctxWithHTTPResp := runtime.WithCaptureHTTPResponse(context.Background(), &httpResp)
+//
+//	err = iaasalphaClient.AddPublicIpToServer(ctxWithHTTPResp, projectId, serverId, publicIpId).Execute()
+//
+//	requestId := httpResp.Header[wait.XRequestIDHeader][0]
+//
+//	_, err = wait.ProjectRequestWaitHandler(context.Background(), iaasalphaClient, projectId, requestId).WaitWithContext(context.Background())
+func ProjectRequestWaitHandler(ctx context.Context, a APIClientInterface, projectId, requestId string) *wait.AsyncActionHandler[iaasalpha.Request] {
```
Comment (joaopalet) [inline services/iaasalpha/wait/wait.go:198]:
```suggestion

// ProjectRequestWaitHandler will wait for a request to succeed.
//
// It receives a request ID that can be obtained from the "X-Request-Id" header in the HTTP response of any operation in the IaaS API.
// To get this response header, use the "runtime.WithCaptureHTTPResponse" method from the "core" packaghe to get the raw HTTP response of an SDK operation.
// Then, the value of the request ID can be obtained by accessing the header key which is defined in the constant "XRequestIDHeader" of this package.
//
// Example usage:
//
//	var httpResp *http.Response
//	ctxWithHTTPResp := runtime.WithCaptureHTTPResponse(context.Background(), &httpResp)
//
//	err = iaasalphaClient.AddPublicIpToServer(ctxWithHTTPResp, projectId, serverId, publicIpId).Execute()
//
//	requestId := httpResp.Header[wait.XRequestIDHeader][0]
//	_, err = wait.ProjectRequestWaitHandler(context.Background(), iaasalphaClient, projectId, requestId).WaitWithContext(context.Background())
func ProjectRequestWaitHandler(ctx context.Context, a APIClientInterface, projectId, requestId string) *wait.AsyncActionHandler[iaasalpha.Request] {
```

Comment (joaopalet) [reply services/iaasalpha/wait/wait.go:198]: Just removed two unneeded empty `// ` from the example

**Pull request merged**